cogito/protocol.go

```diff
@@ -260,7 +260,9 @@ func (src *Source) Validate() error {
 	//
 	// Validate optional fields.
 	//
-	// In this case, nothing to validate.
+	if src.GhHostname != "" {
+		src.GhHostname = normalizeHostname(src.GhHostname)
```

Comment:
Given your current implementation, you could call `normalizeHostname` in all cases as it would leave an empty string as empty string.

Reply:
If it is possible to always call normalizeHostname, it would be indeed better, because one less conditional to put in one's head to understand.